### PR TITLE
SOV-4553 feat: introduce fallback oracle

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const path = require("path");
+
+const func: DeployFunction = async (hre) => {
+    const { getNamedAccounts, deployments: { deploy, get } } = hre;
+    const { deployer } = await getNamedAccounts();
+    const backupPriceFeed = await get("FallbackOracle");
+    await deploy("PriceFeedRevertOnStalePrice", {
+        from: deployer,
+        args: [
+            "0x87d7fd2e0baac4ae61cc7e728647e2f6d80118a0",
```
Comment (tjcloa) [inline deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: this is the main pricefeed, but why is it hardcoded?   
it should only apply n the RSK mainnet and queried from deployments.

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { DeployFunction } from "hardhat-deploy/types";
+const path = require("path");
+
+const func: DeployFunction = async (hre) => {
+    const { getNamedAccounts, deployments: { deploy, get } } = hre;
+    const { deployer } = await getNamedAccounts();
+    const backupPriceFeed = await get("FallbackOracle");
```
Comment (tjcloa) [inline deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: pls rename the FallbackOracle to the DummyFallbackOracle for clarity

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    await deploy("PriceFeedRevertOnStalePrice", {
+        from: deployer,
+        args: [
+            "0x87d7fd2e0baac4ae61cc7e728647e2f6d80118a0",
+            backupPriceFeed.address,
+        ],
+        log: true,
+        skipIfAlreadyDeployed: true,
```
Comment (tjcloa) [inline deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: why skipIfAlreadyDeployed?

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { DeployFunction } from "hardhat-deploy/types";
+const path = require("path");
+
+const func: DeployFunction = async (hre) => {
```
Comment (tjcloa) [inline deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: as fairly noted by @devtective the deployment should replace the logic of the PriceFeed proxy.  
for the RSK mainnet pls print a note to create a SIP to complete the upgrade with the new logic address.

**Review by tjcloa — CHANGES_REQUESTED** [review]
lgtm but for a few improvements related to the code clarity (see inline comments) and deployment script considering that the PriceFeed contract is upgradeable as fairly noted by @devtective

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 4 above)
Comment (stdsure) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 3 above)
Comment (stdsure) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 2 above)
Comment (stdsure) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 1 above)
Comment (stdsure) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

Comment (cwsnt) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 2 above)
Comment (cwsnt) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 3 above)
Comment (cwsnt) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

File: deployment/deploy/7-PriceFeedRevertOnStalePrice.ts (hunk 4 above)
Comment (cwsnt) [reply deployment/deploy/7-PriceFeedRevertOnStalePrice.ts]: Not relevant anymore with the latest changes in this [commit](https://github.com/DistributedCollective/zero-contracts/pull/6/commits/d89d7837edf44a897517a27d13825f8679133fec)

**Review by tjcloa — APPROVED** [review]

**Review by koirikivi — APPROVED** [review]
LGTM.

**Pull request merged**